src/System/Hooks/useCollectorSignals.tsx

```diff
@@ -0,0 +1,148 @@
+import { useMemo } from "react"
+import { graphql, useFragment } from "react-relay"
+import {
+  useCollectorSignals_me$key,
+  useCollectorSignals_me$data,
+} from "__generated__/useCollectorSignals_me.graphql"
+import { useCollectorSignals_artworksConnection$key } from "__generated__/useCollectorSignals_artworksConnection.graphql"
+import { useCollectorSignals_artwork$key } from "__generated__/useCollectorSignals_artwork.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import { useFeatureFlag } from "System/Hooks/useFeatureFlag"
+
+interface SignalResult {
+  partnerOffer?: { endAt?: string | null } | null
+}
+
+interface UseCollectorSignalsResult {
+  [artworkID: string]: SignalResult | undefined
+}
+
+/**
+ * Hook to load and process collector signals for a list of artworks
+ */
+export const useCollectorSignals = ({
+  artworksConnection,
+  me,
+}: {
+  artworksConnection?: useCollectorSignals_artworksConnection$key | null
+  me?: useCollectorSignals_me$key | null
+}): UseCollectorSignalsResult => {
+  const meData = useFragment<useCollectorSignals_me$key>(ME_FRAGMENT, me)
+
+  const artworksData = useFragment(
+    ARTWORKS_CONNECTION_FRAGMENT,
+    artworksConnection
+  )
+  const partnerOfferSignalsEnabled = useFeatureFlag(
+    "emerald_signals-partner-offers"
+  )
+
+  const artworksDataNodes = extractNodes(artworksData)
+
+  const artworkIDs = JSON.stringify(
+    artworksDataNodes.map(artwork => artwork.internalID).sort()
+  )
+
+  const collectorSignals = useMemo(() => {
+    const result = artworksDataNodes.reduce((acc, artworkData) => {
+      acc[artworkData.internalID] = processCollectorSignalsOnArtwork({
+        me: meData,
+        artwork: artworkData,
+        partnerOfferSignalsEnabled,
+      })
+
+      return acc
+    }, {})
+    return result
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [artworkIDs, meData, partnerOfferSignalsEnabled])
+
+  return collectorSignals
+}
+
+/**
+ * Hook to load and process collector signals for a single artwork
+ */
+export const useCollectorSignalsOnArtwork = ({
+  artwork,
+  me,
+}: {
+  artwork: useCollectorSignals_artwork$key
+  me?: useCollectorSignals_me$key | null
+}): SignalResult => {
+  const meData = useFragment<useCollectorSignals_me$key>(ME_FRAGMENT, me)
+  const partnerOfferSignalsEnabled = useFeatureFlag(
+    "emerald_signals-partner-offers"
+  )
+
+  const artworkData = useFragment(ARTWORK_FRAGMENT, artwork)
+
+  const result = processCollectorSignalsOnArtwork({
+    me: meData,
+    artwork: artworkData,
+    partnerOfferSignalsEnabled,
+  })
```

Comment:
Rather than passing this off to another function, can we do the computation inline here? The other function isn't complex enough to warrant breaking it out.

Reply:
The reasoning here is:
- I want to re-use the logic for the second hook which works on a single artwork, not an iterator
- I can't iterate over the second hook because it breaks the rules of hooks

However, now that I have removed the relay dependencies it might make sense to drop the second hook altogether until we need it - maybe easier to pass in `artworks: [artwork]` or something like that. I'll do it for now.

Reply:
Was able to put this all in a single hook with an overloaded signature [here](https://github.com/artsy/force/pull/14132/files#diff-bc2e0d2ca40c967cf46698eaa9a12ae89ae875ce7977cbd4e2cd22d7b46cbd3eR34-R57)